bm-persona/Dining/DiningDetailViewController.swift

```diff
@@ -144,6 +144,7 @@ extension DiningDetailViewController: TabBarControlDelegate {
         self.menuView.setData(data: meals[mealNames[control.index]]!)
         self.menuView.tableView.reloadData()
         self.menuView.filter.deselectAllItems()
+        self.menuView.missingView.isHidden = true
```

Comment:
What if you called `self.menuView.update()` instead? The missing view is updated in that function, and it could be the case that the tab you switch to still has no items, even if the filters are all deselected.